functional/include/fun/option/option_inner.h

```diff
@@ -59,81 +60,105 @@ struct OptionUnion<T, std::enable_if_t<std::is_empty<T>::value>>: private T {
     return true;
   }
 
-  T unwrap() { return static_cast<T&&>(*this); }
+  T dump() {
+    _variant = Tag::NONE;
+    return static_cast<T&&>(*this);
+  }
 
   template <class... Args>
   void emplace(Args&&... args) {
-    static_cast<T&>(*this) = T(std::forward<Args>(args)...);
+    // Constructing a temporary and throwing it away mostly replicates the
+    // observable compile-time behavior of a normal emplace().
+    void(T(std::forward<Args>(args)...));
```

Comment:
What does the `void(...)` part do?

Reply:
It turns the type of the expression to `void`, i.e. ignores the result. That way it's clear to compilers (and hopefully also humans) that I intended to throw away the object I just created.

Reply:
Sounds good